lib/replicache.dart

```diff
@@ -340,10 +343,17 @@ class Replicache implements ReadTransaction {
     }
   }
 
-  void _fireOnChange() {
-    // TODO(arv): We should pass the ref of the current commit to ensure we get
-    // consistent results in all the subscribe callbacks.
-    _changeStreamController.add(null);
+  void _fireOnChange() async {
+    final List<_Subscription> subscriptions = _subscriptions
+        .where((s) => !s.streamController.isPaused)
+        .toList(growable: false);
+    final results = await query((tx) async {
+      final futures = subscriptions.map((s) => s.callback(tx));
```

Comment:
What happens if the callback throws?

Reply:
Look at 0bf4248ebb2910bf19b928cbf6c0328abf7ae6e0. Isn't that beautiful =P